ci/test_notebooks.sh

```diff
@@ -11,7 +11,10 @@ rapids-logger "Generate notebook testing dependencies"
 rapids-dependency-file-generator \
   --output conda \
   --file-key test_notebooks \
-  --matrix "cuda=${RAPIDS_CUDA_VERSION%.*};arch=$(arch);py=${RAPIDS_PY_VERSION}" | tee env.yaml
+  --matrix "cuda=${RAPIDS_CUDA_VERSION%.*};arch=$(arch);py=${RAPIDS_PY_VERSION}"  \
+  --prepend-channel "${CPP_CHANNEL}" \
+  --prepend-channel "${PYTHON_CHANNEL}" \
+| tee env.yaml
```

Comment:
At this point in this script, `CPP_CHANNEL` and `PYTHON_CHANNEL` haven't yet been set. If you want the downloaded CI artifacts to be considered in the conda solve, you'll have to move this block from lower down up above this:

```shell
rapids-logger "Downloading artifacts from previous jobs"
CPP_CHANNEL=$(rapids-download-conda-from-s3 cpp)
PYTHON_CHANNEL=$(rapids-download-conda-from-s3 python)
```

If you do that, then it'd also be good to remove the `rapids-mamba-retry install cugraph-dgl` in favor of that coming through this, so there will be a single call to create the environment.

Here's an example: https://github.com/rapidsai/ucx-py/pull/1101